core/src/main/scala/org/apache/spark/executor/Executor.scala

```diff
@@ -300,8 +300,16 @@ private[spark] class Executor(
               m
             }
           }
-          val taskEndReason = new ExceptionFailure(t, metrics)
-          execBackend.statusUpdate(taskId, TaskState.FAILED, ser.serialize(taskEndReason))
+          val serializedTaskEndReason = {
+            try {
+              ser.serialize(new ExceptionFailure(t, metrics))
```

Comment:
Is it possible that an exception does not throw a NotSerializableException but cannot be deserialized on the other side? This could be due to, say, class loader issues or other issues not caught during serialization time.